### PR TITLE
#184760038 user signUp/register

### PR DESCRIPTION
**Pull request by teerenzo**
### What does this PR do?

to allow user to be able to Register/signup to our site

### Description of Task to be completed?

user will be able allowed to register and he/she will get an email to verify his/her account

### How should this be manually tested?

using deployed version

https://trojans-ec-fe-git-ft-register-signup-184760038-atlp-trojan.vercel.app/register

### Any background context you want to provide?

N/A

### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/184760038

#### Figma
signup2

signup1

#### Actual
Screen Shot 2023-04-06 at 10 57 53
Screen Shot 2023-04-06 at 10 57 09

### Questions:

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code is formatted correctly
- [x] My changes doesn't have any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by NG-Solennel — APPROVED** [review]

**Review by Marthatwesi — APPROVED** [review]

File: .codeclimate.yml [hunk 1]
```diff
@@ -12,17 +12,17 @@ checks:
   file-lines:
     enabled: true
     config:
-      threshold: 250
```
Comment (Eli250) [inline .codeclimate.yml]: Change it back to 250

File: .env [hunk 2]
```diff
@@ -1,2 +1 @@
-node_modules
-
```
Comment (Eli250) [inline .env:2]: Don't push the .env file

File: .env.example [hunk 3]
```diff
@@ -0,0 +1 @@
+BACKEND_URL=https://trojans-ec-bn-staging.onrender.com/api/v1
```
Comment (Eli250) [inline .env.example]: Use dummy variable. Don't expose your API

File: src/assets/javascript/signUp.js [hunk 4]
```diff
@@ -0,0 +1 @@
+console.log("hello")
```
Comment (Eli250) [inline src/assets/javascript/signUp.js]: There is no use for this file.

File: src/pages/signUp.js [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+                    placeHolder="CITY"
+                    onChange={(e) => setCity(e.target.value)}
+                    required
+                    register={{ ...billingRegister("city") }}
+                    errorText={billingErros.city?.message}
+                  />
+                  <Input
+                    type="text"
+                    placeHolder="PROVINCE"
+                    onChange={(e) => setProvince(e.target.value)}
```
Comment (Eli250) [inline src/pages/signUp.js]: Why do you need to use onchange function with register?

File: src/components/signUp/Details.js [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { UserDetailSchema } from "../../schema/signUpSchema";
+import Input from '../input';
+import Button from '../button';
+import { setCurrentStepper,setData } from "../../redux/features/slices/stepper";
+const Details = () => {
```
Comment (Eli250) [inline src/components/signUp/Details.js]: What's the meaning of const Deatails ... in signup?

**Review by Eli250 — CHANGES_REQUESTED** [review]

File: src/components/signUp/Details.js (hunk 6 above)
Comment (teerenzo) [reply src/components/signUp/Details.js]: UserDetails

**Review by Eli250 — APPROVED** [review]

**Pull request merged**